src/functional/swing_slide_filter.zig

```diff
@@ -278,10 +509,54 @@ fn appendValue(comptime T: type, value: T, compressed_values: *std.ArrayList(u8)
     }
 }
 
-/// Computes the intercept coefficient of a linear function that passes through the `point` with
-/// the given `slope` coefficient.
-fn computeInterceptCoefficient(slope: f80, point: DiscretePoint) f80 {
-    return point.value - slope * usizeToF80(point.time);
+/// Computes the intercept coefficient of a linear function that passes through the `DiscretePoint`
+/// `point` with the given `slope` coefficient.
+fn computeInterceptCoefficient(slope: f80, comptime point_type: type, point: point_type) f80 {
+    if (point_type == DiscretePoint) {
+        return point.value - slope * usizeToF80(point.time);
+    } else {
+        return point.value - slope * point.time;
+    }
+}
+
+/// Updates the linear function coeficients in `linear_function` that passes throught the two
+/// points of the `segment` slided up/down at the start point and down/up at the end point based
+/// on the `error_bound`. Specifically, the `linear_function` will pass through the points
+/// (`segment.start_point.time`, `segment.start_point.value - error_bound`) and
+/// (`segment.end_point.time`, `segment.end_point.value + error_bound`).
+fn updateSlideLinearFunction(
+    segment: DiscreteSegment,
+    linear_function: *LinearFunction,
+    error_bound: f32,
+) void {
+    if (segment.end_point.time != segment.start_point.time) {
+        const duration: f80 = @floatFromInt(segment.end_point.time - segment.start_point.time);
+        linear_function.slope = (segment.end_point.value + 2 * error_bound -
+            segment.start_point.value) / duration;
+        linear_function.intercept = segment.start_point.value - error_bound -
+            linear_function.slope * usizeToF80(segment.start_point.time);
+    } else {
+        linear_function.slope = 0.0;
+        linear_function.intercept = segment.start_point.value;
+    }
+}
+
+/// Computes the interception point between `linear_function_1` and `linear_function_2` and
+/// returns it in `point`. If the lines are parallel, the interception is `undefined`.
+fn computeInterceptionPoint(
+    linear_function_1: LinearFunction,
+    linear_function_2: LinearFunction,
+    point: *ContinousPoint,
+) void {
+    if (linear_function_1.slope != linear_function_2.slope) {
+        point.time = @floatCast((linear_function_2.intercept - linear_function_1.intercept) /
+            (linear_function_1.slope - linear_function_2.slope));
+        point.value = @floatCast(linear_function_1.slope * point.time + linear_function_1.intercept);
+    } else {
+        // There is no interception, the linear functions are parallel.
```

Comment:
I am not sure if an error is the best option here. It is possible that the lines are parallel, since they are the same. Instead, I just realized that the interception can be any point, instead of being undefined. I have changed the code to reflect this and avoid undesired behaviour.

Reply:
Ok I though the `else` case meant that an error had occurred and `undefined` was used to report an error. The new code and comment makes it clear that it is an acceptable state.